js/ui/grid_core/ui.grid_core.header_filter_core.js

```diff
@@ -360,6 +357,28 @@ export const HeaderFilterView = modules.View.inherit({
         }
     },
 
+    _normalizeHeaderFilterOptions(options) {
+        const generalHeaderFilter = this.option('headerFilter') || {};
+        const specificHeaderFilter = options.headerFilter || {};
+
+        const generalDeprecated = {
+            search: {
+                enabled: generalHeaderFilter.allowSearch,
+                timeout: generalHeaderFilter.searchTimeout,
+            }
+        };
```

Comment:
Is it expected behavior, that ```search.enabled``` and ```search.timeout``` will be always overridden by ```specificDeprecated``` values?

Reply:
Yes. Example. If `datagrid.headerFilter.search.enabled = true` and in `datagrid.columns[0].headerFilter.search.enabled = false` then in first column's headerFilter search input shouldn't appear

Reply:
Okay, our ```extend``` method has a different behavior that native ```extend```
It's not override values with ```undefinded``` value :)